service/matching/task_queue_internal_info_cache.go

```diff
@@ -0,0 +1,73 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package matching
+
+import (
+	enumspb "go.temporal.io/api/enums/v1"
+	taskqueuespb "go.temporal.io/server/api/taskqueue/v1"
+	"go.temporal.io/server/common/cache"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/tqid"
+)
+
+const (
+	taskQueueInternalInfoCacheMaxSize = 10000
+)
+
+/*
+In-memory Cache for storing task queue child partitions' internal info (Stats, Pollers)
+Stores key-value pairs as: PartitionKey -> physicalInfoByBuildId
+*/
+
+type taskQueueInternalInfoCache struct {
+	cache          cache.Cache
+	metricsHandler metrics.Handler
```

Comment:
hmm, good point about not requiring a metric handler. Removed.

My inspiration for this cache was the one for the reachability since they are simple and similar too. I don't see a harm in creating an interface type, but was wondering:

- Is it good practice to create an interface when we know we have only one implementation class? I was thinking it *could* be an over-kill given the niche use case of this cache but would love to hear your opinions.
- If we do agree on going ahead with an interface type, we should consider switching `reachabilityCache` to an interface type too to remain consistent.

I am for the motion of keeping the cache as a non-interface right now and circling back and converting them respective interfaces in the interest of time.